hdfs3/tests/test_hdfs3.py

```diff
@@ -27,6 +28,7 @@ def hdfs():
 
     if hdfs.exists('/tmp/test'):
         hdfs.rm('/tmp/test')
+    hdfs.disconnect()
```

Comment:
`__del__` is last resort, python encourage of use context managers instead half working destructor. As far as I remember `__del__` working as expected only since 3.4.1